openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java

```diff
@@ -1197,7 +1197,12 @@ private Schema getSchemaDefinition(
                     } else {
                         if (type instanceof TypedElement) {
                             ClassElement classElement = ((TypedElement) type).getType();
-                            Optional<ClassElement> superType = classElement == null ? Optional.empty() : classElement.getSuperType();
+                            Optional<ClassElement> superType;
+                            if (classElement.isInterface() && !classElement.getInterfaces().isEmpty()) {
```

Comment:
What if there are multiple interfaces? Why the first one?

Reply:
Otherwise it will be very complicated 😆  
Just kidding. I missed it and I will take a look :) 
There is also a situation where it can both extend a class and implement interface too. We will need a list there to store all super types I guess.